### PR TITLE
Add umbrella header

### PR DESCRIPTION
**Pull request by jlpiedrahita**
Related issue: https://github.com/facebook/componentkit/issues/46

### CODE REVIEW

File: ComponentKit/ComponentKit.h [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+#import <ComponentKit/CKComponentConstantDecider.h>
+#import <ComponentKit/CKComponentDataSource.h>
+#import <ComponentKit/CKComponentDataSourceOutputItem.h>
+#import <ComponentKit/CKComponentDeciding.h>
+#import <ComponentKit/CKComponentPreparationQueueListener.h>
+#import <ComponentKit/CKComponentPreparationQueueTypes.h>
+#import <ComponentKit/CKComponentProvider.h>
+//Debug
+#import <ComponentKit/CKComponentDebugController.h>
+#import <ComponentKit/CKComponentHierarchyDebugHelper.h>
```
Comment (dstnbrkr) [inline ComponentKit/ComponentKit.h]: Let's exclude the debug headers from the umbrella header

File: Examples/WildeGuess/WildeGuess/Components/InteractiveQuoteComponent.mm [hunk 2]
```diff
@@ -11,12 +11,8 @@
 
 #import "InteractiveQuoteComponent.h"
 
-#import <ComponentKit/CKComponentGestureActions.h>
-#import <ComponentKit/CKComponentScope.h>
+#import <ComponentKit/ComponentKit.h>
 #import <ComponentKit/CKComponentSubclass.h>
-#import <ComponentKit/CKCompositeComponent.h>
-#import <ComponentKit/CKOverlayLayoutComponent.h>
-#import <ComponentKit/CKStackLayoutComponent.h>
```
Comment (dstnbrkr) [inline Examples/WildeGuess/WildeGuess/Components/InteractiveQuoteComponent.mm]: I kind of like using the explicit imports in the example project, feels more explanatory

File: Examples/WildeGuess/WildeGuess/Components/QuoteComponent.h [hunk 3]
```diff
@@ -9,7 +9,7 @@
  * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 
-#import "CKCompositeComponent.h"
+#import <ComponentKit/CKCompositeComponent.h>
```
Comment (dstnbrkr) [inline Examples/WildeGuess/WildeGuess/Components/QuoteComponent.h]: but if we do convert the example project to using the umbrella header, let's do it 100% (all headers too)

**Comment by dstnbrkr** [comment]
Umbrella header looks good to go once debug classes are removed. Example project either doesn't need conversion or needs total conversion.

**Comment by jlpiedrahita** [comment]
Removed the debug headers (good catch). Regarding the example project it feels neat to use only the umbrella header, IMHO use of component classes alone should be explanatory enough.

**Comment by dstnbrkr** [comment]
Thanks @jlpiedrahita

**Pull request closed without merging**